plumbing/format/gitignore/matcher.go

```diff
@@ -0,0 +1,33 @@
+// gitignore package implements scanning for and parsing of gitignore files in a directory
```

Comment:
move this comment to a `doc.go` file, dedicated to the package documentation including the content of the README.md